hypervisor/arch/x86/irq.c

```diff
@@ -421,6 +423,21 @@ static void init_irq_descs(void)
 {
 	uint32_t i;
 
+	/*
+	 * Fill in #CONFIG_MAX_VM_NUM posted interrupt specific irq and vector pairs
+	 * at runtime
+	 */
+	for (i = 0U; i < CONFIG_MAX_VM_NUM; i++) {
+		uint32_t idx = i + NR_STATIC_MAPPINGS_1;
+
+		ASSERT(irq_static_mappings[idx].irq == 0U, "");
+		ASSERT(irq_static_mappings[idx].vector == 0U, "");
+
+		irq_static_mappings[idx].irq = POSTED_INTR_IRQ + i;
+		irq_static_mappings[idx].vector = POSTED_INTR_VECTOR + i;
```

Comment:
C compiler seems unable to generate this kind of code in static array definition. 
If we have compile time generated table, we should revisit this later.

Reply:
Yes, that is the reason I added this code to fill in at runtime.